### DISCUSSION TITLE
Could not generate elastic agents with so-elastic-agent-gen-installers

### DISCUSSION THREAD

**Discussion by Sahale** [Security-Onion-Solutions/securityonion#13125] [2.4] [answered]
### Version

2.4.70

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

16

### RAM

64

### Storage for /

200Gb

### Storage for /nsm

6Tb

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hey!

Faced with a problem: if there are more than 20 Elastic Agent policies in Fleet, there is a possibility that `so-elastic-agent-gen-installers` won't get enrollment API token for endpoint-initial with error `Fleet Host URL, Enrollment Token or Elastic Version empty - exiting...`

The problem is in command:
`ENROLLMENTOKEN=$(curl -K /opt/so/conf/elasticsearch/curl.config -L "localhost:5601/api/fleet/enrollment_api_keys" -H 'kbn-xsrf: true' -H 'Content-Type: application/json'  | jq .list | jq -r -c '.[] | select(.policy_id | contains("endpoints-initial")) | .api_key')`

Could temporary fix that with:
`ENROLLMENTOKEN=$(curl -K /opt/so/conf/elasticsearch/curl.config -L "localhost:5601/api/fleet/enrollment_api_keys?perPage=100" -H 'kbn-xsrf: true' -H 'Content-Type: application/json'  | jq .list | jq -r -c '.[] | select(.policy_id | contains("endpoints-initial")) | .api_key')`

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by defensivedepth** [comment, marked as answer]
Yes, good find. I will get this fixed, here is the issue for tracking it: https://github.com/Security-Onion-Solutions/securityonion/issues/13131
